main.tf

```diff
@@ -0,0 +1,105 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+locals {
+  public_cidr=["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr=["10.0.2.0/24", "10.0.3.0/24"]
+}
+
+resource "aws_subnet" "Public" {
+  count = 2
```

Comment:
This will work only if we have two subnets. What should do if we need more?
Use the length function instead of hardcoding 2.
Here is more info:
https://www.terraform.io/language/meta-arguments/count#when-to-use-for_each-instead-of-count